### DISCUSSION TITLE
Binding a form field with a custom query

### DISCUSSION THREAD

**Discussion by khalidsharif** [Budibase/budibase#12428] [Help] [answered]
Hi Everyone,
I am trying to have a have a page showing data in a table from a custom query (MYSQL) and whenever I change the form field data , another query should be performed after clicking an action button and the table should be updated.

here is what I did but However, .. the table is not getting updated:

![image](https://github.com/Budibase/budibase/assets/50369621/ba1a1a4d-b111-48cc-8a4c-c9cd9a8c0fc1)

**Comment by melohagan** [comment]
Hey @khalidsharif 

Perhaps the *Refresh Data Provider* is triggered before the *Execute Query* has had time to complete. You may need to use sync automations: https://docs.budibase.com/docs/synchronous-automations 

EDIT: Actually your query is doing a select, which would not update the data in your table, therefore refresh data provider won't show anything different.

**Reply by khalidsharif** [reply]
So, what is the ideal solution for my case? I want to show specific records based on input field data. Is it OK to pull all records from custom query and then filter it? though I have more than 6000 record

**Reply by melohagan** [reply, marked as answer]
You can use your select query as a source of your data provider + table. You can also pass in binding values which could be used for filtering.
https://docs.budibase.com/docs/data-sources#custom-queries

You don't need the execute query action.

**Reply by khalidsharif** [reply]
Thanks
